src/server.rs

```diff
@@ -63,12 +102,22 @@ pub async fn run(options: Opt) -> Result<(), Box<dyn Error>> {
             }
         };
 
+        let sni = conn
+            .handshake_data()
+            .unwrap()
```

Comment:
`unwrap()` may result in panic in runtime, I think it is unacceptable for server implementation. Although I am using `unwrap()` anywhere which is my dirty implement when I start to do this project, I still want to make it more elegant in our feature development.

Reply:
These two `unwrap()` will always succeed, according to the documentation.
- `Connection::handshake_data()` is guaranteed to return `Some` after the connection fully estabilshed ([documentation](https://docs.rs/quinn/0.10.2/quinn/struct.Connection.html#method.handshake_data)).
- While `Connection::handshake_data().unwrap()` return dynamic type, it is guaranteed to return `crypto::rustls::HandshakeData` with `rustls`, which we used ([documentation](https://docs.rs/quinn/0.10.2/quinn/struct.Connecting.html#method.handshake_data)).

Of course it is better to handle the failure cases, but we can skip it with no panic.